src/main/java/mekanism/client/render/item/ChemicalFluidBarDecorator.java

```diff
@@ -0,0 +1,116 @@
+package mekanism.client.render.item;
+
+import com.mojang.blaze3d.systems.RenderSystem;
+import com.mojang.blaze3d.vertex.BufferBuilder;
+import com.mojang.blaze3d.vertex.Tesselator;
+import mekanism.api.chemical.Chemical;
+import mekanism.api.chemical.ChemicalStack;
+import mekanism.api.chemical.IChemicalHandler;
+import mekanism.api.math.MathUtils;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.Font;
+import net.minecraft.client.renderer.entity.ItemRenderer;
+import net.minecraft.world.item.ItemStack;
+import net.minecraftforge.client.IItemDecorator;
+import net.minecraftforge.client.extensions.common.IClientFluidTypeExtensions;
+import net.minecraftforge.common.capabilities.Capability;
+import net.minecraftforge.fluids.FluidStack;
+import net.minecraftforge.fluids.FluidUtil;
+import net.minecraftforge.fluids.capability.IFluidHandlerItem;
+
+import java.util.Optional;
+import java.util.function.Predicate;
+
+public class ChemicalFluidBarDecorator implements IItemDecorator {
+
+    private final Capability<? extends IChemicalHandler<?,?>>[] chemicalCaps;
+    private final boolean showFluid;
+    private final Predicate<ItemStack> visibleFor;
+
+    /**
+     * @param showFluid if the fluidcapability should be checked for display, display above chemicalCaps if both are present
+     * @param visibleFor checks if bars should be rendered for the given itemstack
+     * @param chemicalCaps the capabilities to be displayed in order, starting from the bottom
+     */
+    public ChemicalFluidBarDecorator(boolean showFluid, Predicate<ItemStack> visibleFor, Capability<? extends IChemicalHandler<?,?>>... chemicalCaps) {
```

Comment:
Missing space after the `,` in the `IChemicalHandler` generic. I also believe this constructor might need a `@SafeVarargs` but I am not positive about that as I haven't pulled it locally yet to look at